cmd/fs_handlers.go

```diff
@@ -122,21 +219,45 @@ func handleStageDirectory(ctx *cli.Context, ctl *client.Client, root, repoRoot s
 		mpb.AppendDecorators(decor.Percentage()),
 	)
 
+	type stageList struct {
+		local    string
+		repoList []string
+	}
+
 	nWorkers := 20
 	start := time.Now()
-	jobs := make(chan stagePair, nWorkers)
+	jobs := make(chan twins, nWorkers)
 
 	// Start a bunch of workers that will do the actual adding:
 	for idx := 0; idx < nWorkers; idx++ {
 		go func() {
 			for {
-				pair, ok := <-jobs
+				twinsSet, ok := <-jobs
 				if !ok {
 					return
 				}
 
-				if err := ctl.Stage(pair.local, pair.repo); err != nil {
-					fmt.Printf("failed to stage %s: %v\n", pair.local, err)
+				firstToStage := ""
+				for i, repoPath := range twinsSet.repoPaths {
+					if i == 0 {
+						firstToStage = repoPath
+						// First occurrence is staged.
+						// Stage creates all needed parent directories.
+						if err := ctl.Stage(twinsSet.localPath, repoPath); err != nil {
+							fmt.Printf("failed to stage '%s' as '%s': %v\n", twinsSet.localPath, repoPath, err)
```

Comment:
Common guideline is that one outputs warnings and errors to ``stderr`` rather than ``stdout``.

Reply:
stdout -> stderr with d89e161